### PR TITLE
catkin plugin: support rosdep pip dependencies

### PR DESCRIPTION
**Pull request by kyrofa**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

The Catkin plugin uses rosdep to resolve dependencies, however, the only type of dependencies it currently supports are apt-based. This PR fixes LP: [#1683036](https://bugs.launchpad.net/snapcraft/+bug/1683036) by adding support for pip-based dependencies.

### CODE REVIEW

File: snapcraft/plugins/_python/_pip.py [hunk 1]
```diff
@@ -99,9 +98,12 @@ def __init__(self, *, python_major_version, part_dir, install_dir,
             self._python_major_version, stage_dir=self._stage_dir,
             install_dir=self._install_dir)
 
-        self._setup()
```
Comment (sergiusens) [inline snapcraft/plugins/_python/_pip.py]: Good call, this makes testing easier :-)

Comment (kyrofa) [reply snapcraft/plugins/_python/_pip.py]: Yeah. I liked having an init that made sure the class was immediately usable, but then it was impossible to determine whether or not things were setup without... setting them up :stuck_out_tongue: .

File: snapcraft/plugins/_python/_pip.py [hunk 2]
```diff
@@ -124,6 +126,10 @@ def _setup(self):
                 # Now that we have our own pip, reset the python home
                 self._python_home = real_python_home
 
+    def is_setup(self):
```
Comment (sergiusens) [inline snapcraft/plugins/_python/_pip.py]: Given the docstring, this should probably be renamed to `is_pip_setup`

Comment (sergiusens) [reply snapcraft/plugins/_python/_pip.py]: And given what it does, I would just make `_is_pip_installed` public and use that instead

Comment (sergiusens) [reply snapcraft/plugins/_python/_pip.py]: And if so, this is probably missing a test, please double check.

Comment (kyrofa) [reply snapcraft/plugins/_python/_pip.py]: I agree with promoting `_is_pip_installed` to public, but I'd rather call it `is_setup()` so we can have the `setup`/`is_setup` logical pair.

Comment (kyrofa) [reply snapcraft/plugins/_python/_pip.py]: Done.

**Pull request merged**